SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -406,6 +406,57 @@ FSpatialNetGUIDCache::FSpatialNetGUIDCache(USpatialNetDriver* InDriver)
 {
 }
 
+typedef TMap<UObject*, uint32> FSubobjectToOffsetMap;
+
+static FSubobjectToOffsetMap CreateOffsetMapFromActor(AActor* Actor, const FClassInfo& Info)
+{
+	FSubobjectToOffsetMap SubobjectNameToOffset;
+
+	for (auto& SubobjectInfoPair : Info.SubobjectInfo)
+	{
+		UObject* Subobject = StaticFindObjectFast(UObject::StaticClass(), Actor, SubobjectInfoPair.Value->SubobjectName);
+		uint32 Offset = SubobjectInfoPair.Key;
+
+		if (Subobject != nullptr && Subobject->IsPendingKill() == false && Subobject->IsSupportedForNetworking())
+		{
+			SubobjectNameToOffset.Add(Subobject, Offset);
+		}
+	}
+
+	if (Actor->GetInstanceComponents().Num() > 0)
+	{
+		// Process components attached to this object; this allows us to join up
+		// server- and client-side components added in the level.
+		TArray<UActorComponent*> ActorInstanceComponents = Actor->GetInstanceComponents();
+
+		// These need to be ordered in case there are more than one component of the same type, or
+		// we may end up with wrong component instances having associations between them.
+		ActorInstanceComponents.Sort([](const UActorComponent& Lhs, const UActorComponent& Rhs) -> bool {
```

Comment:
Is it an issue that Sort is not stable? E.g. is it ok to have arbitrary ordering of components with duplicate names (if that is at all possible)?

Reply:
I don't think you can have multiple objects with the same name on different subobjects, but I'll check this out.

Reply:
Unreal doesn't allow multiple subobjects with the same name.